### DISCUSSION TITLE
Opcua Operations Limit

### DISCUSSION THREAD

**Discussion by xyzakdkd** [FreeOpcUa/opcua-asyncio#1548] [Q&A] [answered]
Hello, i have an Error (specifically, opcua.ua.uaerrors._auto.BadNotSupported) i know that the server only process 50 operations per second .when executing my code. This issue arises due to the numerous requests made at the beginning of the script. I'm seeking guidance on how to modify my script to obtain the first 70 nodes just once during the initial script execution, even as it enters the subscription phase. How can I achieve this? 
```
#list with 70 node_id
node_id = [""]

#list with 43 node_id
node_id_subscription = []

values =[]
    async with client:
        while True:
                first_70_nodes =  await opcua_client.get_values(node_id)
                print (first)

                #subscription
                handler = subhandler.SubHandler()
                subscription = await client.create_subscription(500, handler)
                handler.subscription = subscription
                for node_path in node_id_subscription:
                    node_ids = client.get_node(node_id_subscription)
                    await subscription.subscribe_data_change(node_ids)
            await asyncio.sleep(1)

if __name__ == "__main__":
    asyncio.run(task())


### in another class called  ###
  async def get_values(self, node_ids):
      list=[]
      for sublist in node_ids:
          for node_id in sublist:
              time.sleep(0.5)
              node = self.client.get_node(node_id)
              data = node.get_data_value()
              list.append([str(node), str(data)])
              
      return list


```

**Comment by AndreasHeine** [comment]
for subscribing large numbers of nodes:

https://github.com/AndreasHeine/opcua-tutorial/blob/8f1dc98e8f2301514aff5fd83ea31fdb35c82029/client/client1.py#L34

for reading multiple nodes at once:
https://github.com/FreeOpcUa/opcua-asyncio/blob/b9e2af4e8d079b3a0c4eb6a6e49f861e8e309fb7/asyncua/client/client.py#L906

**Reply by xyzakdkd** [reply]
I apologize for any confusion. To clarify, the first 70 nodes represent static statuses that remain unchanged throughout the entire process. My intention is to capture these nodes value. it doesnt matter if i got all 70 nodes value  in 10 seconds or 20 seconds but not longer. because they are remain unchanged.

**Reply by AndreasHeine** [reply, marked as answer]
then use "read_values(node_list)" -> it creates just one request/response and is most efficient!

currently you ask in a for loop -> request/response x length of your list which is inefficient

**Reply by xyzakdkd** [reply]
ok i will try it. thanks

**Reply by xyzakdkd** [reply]
Thank you @AndreasHeine  i works with me
